torch/nn/parallel/distributed.py

```diff
@@ -957,6 +961,23 @@ def no_sync(self):
         finally:
             self.require_backward_grad_sync = old_require_backward_grad_sync
 
+    @classmethod
+    def get_active_ddp_module(cls):
+        """
+        TorchDynamo needs to know whether DDP is currently active, and access the DDP module in order to cooperatively optimize it.
+        """
+        if hasattr(cls._tls_ctx, '_active_ddp_module') and cls._tls_ctx._active_ddp_module is not None:
+            return cls._tls_ctx._active_ddp_module
+        return None
+
+    @contextmanager
+    def _inside_ddp_forward(self):
```

Comment:
consider explicitly registering this as dynamo skip or wrapping with dynamo disable?

Reply:
good point, I do want to skip dynamo in this frame and only enter dynamo for the `module_to_run` frame below. I will add that here.